File: src/Census.Contracts/Contracts/CompletedCensusDto.cs
using System;
using System.ComponentModel.DataAnnotations;
using Census.Contracts.Validation.Attributes;

namespace Census.Contracts.Contracts
{
    public class CompletedCensusDto
    {
        [Required]
        [NotEmpty]
        public Guid? Id { get; set; }

        [Required]
        [MagicCensusToken]
        public string AccessToken { get; set; }

        [Required]
        public string LegalName { get; set; }

        [Required]
        public string BaristaName { get; set; }

        [Required]
        [Range(1, 300)]
        public int? BeardLength { get; set; }

        [Required]
        [Range(100, 200)]
        public int? GearInches { get; set; }

        [Required]
        [Range(5, 120)]
        public int? BeerBitterness { get; set; }

        [Required]
        public string FavouriteBand { get; set; }
    }
}

File: src/census/src/infrastructure/api/CompletedCensusDto.js
import ArgumentNullError from '../../errors/ArgumentNullError';

class CompletedCensusDto {
    constructor(id, accessToken, legalName, baristaName, beardLength, gearInches, beerBitterness, favouriteBand) {
        if (!id) throw new ArgumentNullError(id);
        if (!accessToken) throw new ArgumentNullError(accessToken);
        if (!legalName) throw new ArgumentNullError(legalName);
        if (!baristaName) throw new ArgumentNullError(baristaName);
        if (!beardLength) throw new ArgumentNullError(beardLength);
        if (!gearInches) throw new ArgumentNullError(gearInches);
        if (!beerBitterness) throw new ArgumentNullError(beerBitterness);
        if (!favouriteBand) throw new ArgumentNullError(favouriteBand);

        this.id = id;
        this.accessToken = accessToken;
        this.legalName = legalName;
        this.baristaName = baristaName;
        this.beardLength = beardLength;
        this.gearInches = gearInches;
        this.beerBitterness = beerBitterness;
        this.favouriteBand = favouriteBand;
    }
}

export default CompletedCensusDto;
